framework/src/org/checkerframework/framework/qual/TargetLocations.java

```diff
@@ -0,0 +1,40 @@
+package org.checkerframework.framework.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A meta-annotation that restricts the type-use locations where a type
+ * qualifier may be written.  When written together with
+ * {@code @Target({ElementType.TYPE_USE})}, the given type qualifier may be
+ * written only at locations listed in the {@code @TargetLocations(...)}
+ * meta-annotation.  The absence of {@code @TargetLocations(...)}
```

Comment:
Let's make this clearer by writing:

{@code @Target({ElementType.TYPE_USE})} together with no {@code @TargetLocations(...)} means that...